### PR TITLE
Doc 2783 dynamic infra provisioners

### PR DESCRIPTION
**Pull request by michaelcretzman**
# Topics to review

## Provisioner topics:

- [Provisioning overview](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/cd-infrastructure/provisioning-overview)
- [Terraform provisioning overview](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/cd-infrastructure/terraform-infra/terraform-provisioning-with-harness)
- [Provision target deployment infrastructure dynamically with CloudFormation](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/cd-infrastructure/cloudformation-infra/provision-target-deployment-infra-dynamically-with-cloud-formation)
- [CloudFormation provisioning with Harness](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/cd-infrastructure/cloudformation-infra/cloud-formation-provisioning-with-harness)
- [Provision target deployment infrastructure dynamically with CloudFormation](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/cd-infrastructure/cloudformation-infra/provision-target-deployment-infra-dynamically-with-cloud-formation)
- [Azure Resource Management (ARM)](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/cd-infrastructure/azure-arm-provisioning)
- [Azure Blueprint](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/cd-infrastructure/azure-blueprint-provisioning)
- [Shell Script Provisioning](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/cd-infrastructure/shell-script-provisioning)

## Swimlane topics:

Look for the **Define the infrastructure** heading in each topic:

- [Kubernetes infrastructure](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/define-your-kubernetes-target-infrastructure)
- [Deploy Azure Web Apps](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/deploy-srv-diff-platforms/azure/azure-web-apps-tutorial)
- [Deploy Azure ACR to Azure AKS](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/deploy-srv-diff-platforms/azure/azure-cd-quickstart)
- [ECS deployments overview](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial)
- [AWS Lambda](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/deploy-srv-diff-platforms/aws/aws-lambda-deployments)
- [Google Cloud Functions](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/deploy-srv-diff-platforms/google-functions)
- [Serverless.com framework for AWS Lambda](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/deploy-srv-diff-platforms/serverless-lambda-cd-quickstart)
- [Tanzu Application Services deployments overview](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart)
- [VM deployments using SSH](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/deploy-srv-diff-platforms/traditional/ssh-ng)
- [Windows VM deployments using WinRM](https://64d12ce525e28c0aa6e71a8c--harness-developer.netlify.app/docs/continuous-delivery/deploy-srv-diff-platforms/traditional/win-rm-tutorial)

## What Type of PR is This?

- [x ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* https://harness.atlassian.net/browse/DOC-2783

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by thisrohangupta — DISMISSED** [review]

File: docs/continuous-delivery/deploy-srv-diff-platforms/serverless-lambda-cd-quickstart.md [hunk 1]
```diff
@@ -440,6 +451,144 @@ Now that you have configured the Service, we can define the target for our deplo
...
+- Terraform Cloud
+- CloudFormation
+- Shell Script
+
+#### Adding dynamic provisioning to the stage
+
+To add dynamic provisioning to a Harness pipeline Deploy stage, do the following:
+
+1. In a Harness Deploy stage, in **Environment**, enable the option **Provision your target infrastructure dynamically during the execution of your Pipeline**.
+2. Select the type of provisioner do you want to use.
```
Comment (douglas-j-bothwell) [inline docs/continuous-delivery/deploy-srv-diff-platforms/serverless-lambda-cd-quickstart.md:490]:
```suggestion
2. Select the type of provisioner that you want to use.
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md [hunk 2]
```diff
@@ -366,6 +378,127 @@ The target space is your TAS space. This is where you will deploy your applicati
...
+
+Currently, the dynamic provisioning documented in this topic is behind the feature flag `CD_NG_DYNAMIC_PROVISIONING_ENV_V2`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
+
+:::
+
+Here is a summary of the steps to dynamically provision the target infrastructure for a deployment:
+
+1. **Add dynamic provisioning to the CD stage**:
+   1. In a Harness Deploy stage, in **Environment**, enable the option **Provision your target infrastructure dynamically during the execution of your Pipeline**.
+   2. Select the type of provisioner do you want to use.
```
Comment (douglas-j-bothwell) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:393]:
```suggestion
   2. Select the type of provisioner that you want to use.
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/traditional/ssh-ng.md [hunk 3]
```diff
@@ -113,96 +113,269 @@ In **Artifact Details**, enter the following:
...
+```
+
+
+In the Harness Infrastructure Definition, you map outputs to their corresponding settings using expressions in the format `<+provisioner.OUTPUT_NAME>`, such as `<+provisioner.region>`.
+
+<figure>
+
+<docimage path={require('./static/8722541e819fd752abc35693bd00e38cca2bce5df264afa89cbf61288fbc0604.png')} width="60%" height="60%" title="Click to view full size image" />
+
+<figcaption>Figure: Mapped outputs.</figcaption>
```
Comment (douglas-j-bothwell) [inline docs/continuous-delivery/deploy-srv-diff-platforms/traditional/ssh-ng.md:363]:
```suggestion
<figcaption>Figure: Mapped outputs</figcaption>
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/traditional/win-rm-tutorial.md [hunk 4]
```diff
@@ -148,11 +171,211 @@ We will now create the credentials for the secret that is used by Harness to con
...
+
+Currently, the dynamic provisioning documented in this topic is behind the feature flag `CD_NG_DYNAMIC_PROVISIONING_ENV_V2`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
+
+:::
+
+Here is a summary of the steps to dynamically provision the target infrastructure for a deployment:
+
+1. **Add dynamic provisioning to the CD stage**:
+   1. In a Harness Deploy stage, in **Environment**, enable the option **Provision your target infrastructure dynamically during the execution of your Pipeline**.
+   2. Select the type of provisioner do you want to use.
```
Comment (douglas-j-bothwell) [inline docs/continuous-delivery/deploy-srv-diff-platforms/traditional/win-rm-tutorial.md:186]:
```suggestion
   2. Select the type of provisioner that you want to use.
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/traditional/win-rm-tutorial.md [hunk 5]
```diff
@@ -148,11 +171,211 @@ We will now create the credentials for the secret that is used by Harness to con
...
+```
+
+
+In the Harness Infrastructure Definition, you map outputs to their corresponding settings using expressions in the format `<+provisioner.OUTPUT_NAME>`, such as `<+provisioner.region>`.
+
+<figure>
+
+<docimage path={require('./static/8722541e819fd752abc35693bd00e38cca2bce5df264afa89cbf61288fbc0604.png')} width="60%" height="60%" title="Click to view full size image" />
+
+<figcaption>Figure: Mapped outputs.</figcaption>
```
Comment (douglas-j-bothwell) [inline docs/continuous-delivery/deploy-srv-diff-platforms/traditional/win-rm-tutorial.md:358]:
```suggestion
<figcaption>Figure: Mapped outputs</figcaption>
```

**Review by douglas-j-bothwell — APPROVED** [review]
A number of typos and sugs but otherwise looks good.

**Pull request merged**